Add support for PTL + NPCX4 200 pins card

Asthachavda opened the pull request:
Add support for PTL + NPCX4

Asthachavda committed:
boards: nuvoton: Add GPIO configuration for PTL + NPCX 200 pins

Add support for PTL + NPCX4

Signed-off-by: Astha Chavda <astha.chavda@intel.com>

albertofloyd committed:
app: saf: Decouple SAF from Zephyr SPI driver API

Allow no SPI initialization via SPI driver sequences
Move SPI flash initialization in separate file
Add app saf module CMakelist to allow library in the future

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

albertofloyd committed:
drivers: spi opcodes: Add Macronix opcodes

Add Macronix opcodes

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

albertofloyd committed:
oot: mtls: Add alias for eSPI TAF

No all EC vendors name their HW blocks the same way
Introduce use of alias to simplify device mapping.

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

Asthachavda committed:
oot: npcx4_ptl: Add board support for PTL + NPCX4

Add board files for PTL + NPCX4

Signed-off-by: Astha Chavda <astha.chavda@intel.com>

Asthachavda committed:
drivers: add GPIO, FAN & ACPI for PTL + NPCX4

Add required drivers for NPCX4

Signed-off-by: Astha Chavda <astha.chavda@intel.com>

Asthachavda committed:
readme: add missing defination for npcx clock in zephyr patch

Signed-off-by: Astha Chavda <astha.chavda@intel.com>

albertofloyd commented on out_of_tree_boards/boards/arm/npcx4m8f_ptl/npcx4m8f_ptl.dts:
```diff
@@ -21,8 +21,6 @@
 	aliases {
 		i2c0 = &i2c2_0;
 		i2c1 = &i2c0_0;
-		ssd-i3c-bus = &i3c2;
```
@Asthachavda  you should squash this commit with https://github.com/intel/ecfw-zephyr/pull/78/commits/86dbc4a13553195f20b719e4d28436789558ee54

albertofloyd commented on boards/npcx4m8f_ptl.conf:
```diff
@@ -1,20 +1,5 @@
 # EC FW features configuration
 # ----------------------------
-# Battery support
```
This commit should be squash into 843ff107fb4c75d655cc7f908cbedf07b24b8476

Asthachavda replied:
Addressed

albertofloyd reviewed (changes requested):
Needs to rebase and squash commits indicated

Asthachavda committed:
drivers: espi : Enable interrupt for NPCX

For NPCX espi driver interrupt is disable by default

Signed-off-by: Astha Chavda <astha.chavda@intel.com>

Asthachavda committed:
app: debug: Handle eSPI driver multi-byte

Some EC vendors provide multi-byte payload

Signed-off-by: Astha Chavda <astha.chavda@intel.com>

Asthachavda committed:
misc: plt_npcx: rename spi_image to ksc

- rename binary file name to ksc.
- Removed deprecated RVP board references from build scripts.

Signed-off-by: Astha Chavda <astha.chavda@intel.com>

Asthachavda committed:
misc: Increase Minor version to 4.01.00.00

Update the minor EC FW version

Signed-off-by: Astha Chavda <astha.chavda@intel.com>

Commit:
readme : Modify for applying patches for zephyr 3.6

- add the patch require for zephyr v3.6

Signed-off-by: astha <asthacha@nuc7.iind.intel.com>
